### PR TITLE
#196: Edit Architecture: Beginner: Different types of Architectures

### PR DESCRIPTION
**Pull request by gabriellsesam**
Please give feedback on my edit of this topic.
Some things to question on this pr:
* Is the intent of what you wrote intact after my edit?
* Do the changes make sense?
* Is there something you would like to see added or removed?

### CODE REVIEW

**Review by JonasAls89 — CHANGES_REQUESTED** [review]
Lines 14-15 shifts the meaning compared to what I intended. In addition, I think the wording on line 15 is suboptimal.

Line 17-18: I don't think it's correct to say: "we want to lay the foundation...". I would rephrase.

Line 25: Are you sure it's the simplest integration principle? 

Line 29: It's more consistent to write system A and system B.

Line 30: Typo: "threshold"

Line 81: I dont think it's correct to state: "these systems store".. store is not necessarily what DDA focuses on.

Line 96-97: I think you have to add "albeit flexible", as robust and flexible contradicts one another.

Line 104: Does not make sense. Needs to be rewritten.

**Comment by gabriellsesam** [comment]
> Lines 14-15 shifts the meaning compared to what I intended. In addition, I think the wording on line 15 is suboptimal.
> 

Data is not by itself mobile and would therefore be inprecice to say that DDA. DDA gives data mobility, not the other way around. However, I think we can keep this point somehow by reprhasing as:
`Recently Data Driven Architecture has mobilized data in new ways by placing data and its meaning at it's center.`

> Line 17-18: I don't think it's correct to say: "we want to lay the foundation...". I would rephrase.
> 

The edit says lay out the history, but we could rephrase it like
`To show the evolution of Integration Architectures and why Data Driven Architecture is the natural step forward, we will first introduce you to P2P and ESB.`

> Line 25: Are you sure it's the simplest integration principle?
> 

Yes, as in math a point is the simplest type of object based on amount of properties to contend with. A line would just be a connection of two points (P2P)- which in turn is simpler than a triangle, rectangle and so on.

> Line 29: It's more consistent to write system A and system B.
> 

agree

> Line 30: Typo: "threshold"
> 

threefold is correct here, as we go from 1 to 3 connections by adding system C.

> Line 81: I don't think it's correct to state: "these systems store".. store is not necessarily what DDA focuses on.
> 

I'm not sure what you mean as the new phrasing does not say we store the data, but rather focus on the data which these systems store instead of focusing on the system itself. I think it is more precise.

> Line 96-97: I think you have to add "albeit flexible", as robust and flexible contradicts one another.
> 

Can be rephrased to `you need logical and robust principles to create flexible data flows and models`

> Line 104: Does not make sense. Needs to be rewritten.

We can remove line 102-104

**Review by JonasAls89 — APPROVED** [review]

File: training/010_architecture_and_concepts/020_Beginner.rst [hunk 1]
```diff
@@ -9,27 +9,32 @@ Different types of Architectures
...
-another system is introduced and so multiple new messages need to be
-sent? Well, in this case P2P is not your best option, and the issue of
-scalability presents itself. To illustrate how complexity can make P2P a
+This kind of IA is the simplest integration principle. As an example of P2P
+imagine sending a message generated in one System A to a single receiver
+in another System B. System A talks to System B.
+It is simple and therefore easy to manage, albeit what if another System C is
+introduced and needs to talk with both System A and System B?
+Introducing System C will increase the complexity of the architecture threefold,
+as we go from 1 integration between System A and System B to 3 integrations.
```
[the remark this answers is not in the document]
Comment (gabriellsesam) [reply training/010_architecture_and_concepts/020_Beginner.rst]: La dette til under "Semantisk Veiledning" i onenote / Kursutvikling/ Retningslinjer

**Review by ga-hegsvold — APPROVED** [review]
